packages/js/product-editor/src/blocks/product-fields/regular-price/edit.tsx

```diff
@@ -96,34 +98,38 @@ export function Edit( {
 	}, [] );
 
 	return (
-		<div { ...blockProps }>
-			<BaseControl
-				id={ regularPriceId }
-				help={
-					regularPriceValidationError
-						? regularPriceValidationError
-						: renderHelp()
-				}
-				className={ classNames( {
-					'has-error': regularPriceValidationError,
-				} ) }
-			>
-				<InputControl
-					{ ...inputProps }
+		<>
+			<div { ...blockProps }>
+				<BaseControl
 					id={ regularPriceId }
-					name={ 'regular_price' }
-					ref={ regularPriceRef }
-					label={
-						tooltip ? (
-							<Label label={ label } tooltip={ tooltip } />
-						) : (
-							label
-						)
+					help={
+						regularPriceValidationError
+							? regularPriceValidationError
+							: renderHelp()
 					}
-					disabled={ disabled }
-					onBlur={ validateRegularPrice }
-				/>
-			</BaseControl>
-		</div>
+					className={ classNames( {
+						'has-error': regularPriceValidationError,
+					} ) }
+				>
+					<InputControl
+						{ ...inputProps }
+						id={ regularPriceId }
+						name={ 'regular_price' }
+						ref={ regularPriceRef }
+						label={
+							tooltip ? (
+								<Label label={ label } tooltip={ tooltip } />
+							) : (
+								label
+							)
+						}
+						disabled={ disabled }
+						onBlur={ validateRegularPrice }
+					/>
+				</BaseControl>
+			</div>
+			<RegularPriceMenuGroup />
+			<RegularPriceMultipleSelectionMenuItem />
```

Comment:
I love how easily this ties the menu items to the regular price field.

But, I am concerned that this depends on the field always being rendered, even if it isn't visible.

I wonder if we should instead go with a solution where the menu has an optional `templateBlockId` prop that can be set and then it could use a hook like `useHasTemplateBlock( id )` to conditionally render itself.

We should see what the rest of the team thinks about this. Again, I love the simplicity of this, but it will break if we don't always render all blocks in the template.